codebase/che/client.go

```diff
@@ -220,6 +220,108 @@ func (cs *StarterClient) StartExistingWorkspace(ctx context.Context, workspaceNa
 	return &workspaceResp, nil
 }
 
+// GetCheServerState get che server state.
+func (cs *StarterClient) GetCheServerState(ctx context.Context) (*CheServerStateResponse, error) {
+	req, err := http.NewRequest("GET", cs.targetURL("server"), nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	cs.setHeaders(ctx, req)
+
+	if log.IsDebug() {
+		b, _ := httputil.DumpRequest(req, true)
+		log.Debug(ctx, map[string]interface{}{
+			"request": string(b),
+		}, "request object")
+	}
+
+	resp, err := cs.client.Do(req)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated && resp.StatusCode != http.StatusOK {
+		statusErr := CheStarterError{}
+		err = json.NewDecoder(resp.Body).Decode(&statusErr)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"err": err,
+			}, "failed to decode error response from get che server state")
+			return nil, err
+		}
+		log.Error(ctx, map[string]interface{}{
+			"err": statusErr.String(),
+		}, "failed to execute get che server state")
+		return nil, &statusErr
+	}
+
+	cheServerStateResponse := CheServerStateResponse{}
+	err = json.NewDecoder(resp.Body).Decode(&cheServerStateResponse)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to decode response from getting che server state")
+		return nil, err
+	}
+	return &cheServerStateResponse, nil
+}
+
+// StartCheServer start che server if not running.
+func (cs *StarterClient) StartCheServer(ctx context.Context) (*CheServerStateResponse, error) {
+	req, err := http.NewRequest("PATCH", cs.targetURL("server"), nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	cs.setHeaders(ctx, req)
+
+	if log.IsDebug() {
+		b, _ := httputil.DumpRequest(req, true)
+		log.Debug(ctx, map[string]interface{}{
+			"request": string(b),
+		}, "request object")
```

Comment:
same as above, add some extra information in the message to identify the operation.